embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/logs/EmbraceExtensions.kt

```diff
@@ -0,0 +1,120 @@
+package io.embrace.android.embracesdk.internal.logs
+
+import io.embrace.android.embracesdk.LogExceptionType
+import io.opentelemetry.api.common.AttributeKey
+import io.opentelemetry.api.logs.LogRecordBuilder
+
+/**
+ * Extension functions and constants to augment the core OpenTelemetry SDK and provide Embrace-specific customizations
+ *
+ * Note: there's no explicit tests for these extensions as their functionality will be validated as part of other tests.
+ */
+
+/**
+ * Prefix added to all [LogRecordBuilder] attribute keys for all attributes added by the SDK
+ */
+private const val EMBRACE_ATTRIBUTE_NAME_PREFIX = "emb."
+
+/**
+ * Attribute name for the application state (foreground/background) at the time the log was recorded
+ */
+private const val APP_STATE_ATTRIBUTE_NAME = EMBRACE_ATTRIBUTE_NAME_PREFIX + "state"
+
+/**
+ * Attribute name for the id of the session in progress ]at the time the log was recorded
+ */
+private const val SESSION_ID_ATTRIBUTE_NAME = EMBRACE_ATTRIBUTE_NAME_PREFIX + "session_id"
+
+/**
+ * Attribute name for a unique id identifying the log
+ */
+private const val EVENT_ID_ATTRIBUTE_NAME = EMBRACE_ATTRIBUTE_NAME_PREFIX + "event_id"
+
+/**
+ * Attribute name for the exception type in a log representing an exception
+ */
+private const val EXCEPTION_TYPE_ATTRIBUTE_NAME = EMBRACE_ATTRIBUTE_NAME_PREFIX + "exception_type"
+
+/**
+ * Attribute name for the exception name in a log representing an exception
+ */
+private const val EXCEPTION_NAME_ATTRIBUTE_NAME = EMBRACE_ATTRIBUTE_NAME_PREFIX + "exception_name"
+
+/**
+ * Attribute name for the exception message in a log representing an exception
+ */
+private const val EXCEPTION_MESSAGE_ATTRIBUTE_NAME = EMBRACE_ATTRIBUTE_NAME_PREFIX + "exception_message"
+
+/**
+ * Attribute name for the exception context in a log representing an exception
+ */
+private const val EXCEPTION_CONTEXT_ATTRIBUTE_NAME = EMBRACE_ATTRIBUTE_NAME_PREFIX + "exception_context"
+
+/**
+ * Attribute name for the exception library in a log representing an exception
+ */
+private const val EXCEPTION_LIBRARY_ATTRIBUTE_NAME = EMBRACE_ATTRIBUTE_NAME_PREFIX + "exception_library"
+
+/**
+ * Add an attribute representing the application state (foreground/background) at the time the log was recorded
+ */
+internal fun LogRecordBuilder.setAppState(appState: String): LogRecordBuilder {
```

Comment:
Could we model this as an enum instead, and then convert it to the string description within this function?

Reply:
I will make it in a separate PR since it affects the current clients of `MetadataService`.